libs/server-sent-events/src/client.cpp

```diff
@@ -589,11 +589,22 @@ std::shared_ptr<Client> Builder::build() {
     request.set(http::field::host, host);
     request.target(uri_components->encoded_target());
 
+    if (uri_components->has_scheme()) {
+        if (!(uri_components->scheme() == "http" ||
+              uri_components->scheme() == "https")) {
+            return nullptr;
+        }
+    }
+
+    // The resolver accepts either a port number or a service name. If the
+    // URL specifies a port, use that - otherwise, pass in the scheme as the
+    // service name (which will be either http or https due to the check
+    // above.)
     std::string service = uri_components->has_port() ? uri_components->port()
                                                      : uri_components->scheme();
```

Comment:
The issue here was that if a port was provided, say 9090, that's `!= "https"` so it'd end up being a plain http connection.

Reply:
Does this affect any other connections? Events, polling?

Reply:
Looks like it affected our AsioRequester which is used by Events and Polling. I've updated it there as well.